src/differentials/notimplemented.jl

```diff
@@ -0,0 +1,76 @@
+"""
+    NotImplemented
+
+This differential indicates that the derivative is not implemented.
+"""
+struct NotImplemented{M,S,I} <: AbstractDifferential
+    mod::M
```

Comment:
```suggestion
    module::M
```

Reply:
is that allowed?
I thinke `module` is a reserved keyword

Reply:
Oh, I didn't check

Reply:
Yes, that's why I used the name `mod` here. E.g., it is not possible to create the following struct in the REPL
```julia
julia> struct A{M}
          module::M
       end
```
but it works fine if you replace `module` with `mod`.